server/server.go

```diff
@@ -874,17 +877,16 @@ func (s *Server) setSystemAccount(acc *Account) error {
 	acc.mu.Unlock()
 
 	s.sys = &internal{
-		account:  acc,
-		client:   s.createInternalSystemClient(),
-		seq:      1,
-		sid:      1,
-		servers:  make(map[string]*serverUpdate),
-		replies:  make(map[string]msgHandler),
-		sendq:    make(chan *pubMsg, internalSendQLen),
-		statsz:   eventsHBInterval,
-		orphMax:  5 * eventsHBInterval,
-		chkOrph:  3 * eventsHBInterval,
-		eventids: nuid.New(),
```

Comment:
You could have left it in this structure since I believe that we protect access to these fields with the server lock itself.

Reply:
I can move it if you prefer, but personally I stick it nearest to where its used

Reply:
If in same func let's avoid grabbing lock again for getting nuid.

Reply:
no additional locks now, I think it's gtg

Reply:
No big deal. But if you think about it, where we now capture the event ID, we are checking eventsEnabled() which accesses s.sys, s.sys.client and s.sys.account. Those are in the internal structure (again protected by server lock).
So to me if eventids makes more sense in the context of internal, then I would have it in that structure, but if it is going to be used outside then sure, in server struct is fine. We could actually make it very generic and be `snuid` or something that if later we need to get a new NUID under server lock we could use that.
Again, no big deal.

Reply:
I would like to use this same one in JS and other places - though getting at server from a consumer is a bit of a journey, but less than to the extra object below it :)

Reply:
`internal` is only set when we have a system account - so if I want to use it elsewhere it's probably best on server

Reply:
All advisories and all of JS require system accounts and am seriously considering making them on by default for 2.2.0

Reply:
OK, well I dont mind moving it, no big deal, so if you prefer I will, we won't use it now in JS as getting to this would have to go via account which requires locks and stuff to use safely at present

Reply:
I would make one per JS enabled account probably. I think that would be the least amount of contention.

Reply:
How did we leave this one?